idx/memory/memory_test.go

```diff
@@ -925,6 +967,72 @@ func testSingleNodeMetric(t *testing.T) {
 	ix.AddOrUpdate(mkey, data, getPartition(data))
 }
 
+// TestMemoryIndexHeapUsageWithTags5 gives a rough estimate of how much memory the index is using with tag support turned on
+func TestMemoryIndexHeapUsageWithTags5(t *testing.T) {
+	testMemoryIndexHeapUsageWithTags(t, false, 5)
+}
+
+// TestMemoryIndexHeapUsageWithTagsAndUnique5 gives a rough estimate of how much memory the index is using with tag support turned on
+func TestMemoryIndexHeapUsageWithTagsAndUnique5(t *testing.T) {
+	testMemoryIndexHeapUsageWithTags(t, true, 5)
+}
+
+func testMemoryIndexHeapUsageWithTags(t *testing.T, uniqueTags bool, num int) {
+	// turn partitioning off
+	Partitioned = false
+
+	// turn tag support on
+	TagSupport = true
+
+	ix := New()
+	ix.Init()
+
+	tags := []string{
+		"mykey=myvalue",
+		"secondkey=anothervalue",
+		"thirdkey=onemorevalue",
+		"region=west",
+		"os=ubuntu",
+		"anothertag=somelongervalue",
+		"manymoreother=lotsoftagstointern",
+		"afewmoretags=forgoodmeasure",
+		"onetwothreefourfivesix=seveneightnineten",
+		"lotsandlotsoftags=morefunforeveryone",
+		"goodforpeoplewhojustusetags=forbasicallyeverything",
+	}
+	series := getMetricDataWithCustomTags(1, 2, num, 10, "somekindof.longereven.metricname.butinstead.ofashorterone.bunchofthingsandstuff", true, tags, uniqueTags)
+
+	t.Log("Initial memory stats after series has been allocated")
+	printMemUsage(t)
+
+	for _, data := range series {
+		mkey, err := schema.MKeyFromString(data.Id)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ix.AddOrUpdate(mkey, data, 0)
+	}
+
+	t.Log("Memory stats after AddOrUpdate called")
+	printMemUsage(t)
+
+	runtime.GC()
+	time.Sleep(3 * time.Second)
```

Comment:
how did you conclude that this should wait exactly `3` seconds? was that based on trial & error? if there's more behind the number then i think a comment with an explanation would be useful, so somebody else knows what that wait is good for

Reply:
There is nothing more behind the number, it could probably be 1 second.

Reply:
You explained to me that `.GC()` is an async call. So it needs to sleep because we want the GC run to complete before continuing, right? I'd just add a comment like `wait for GC run to complete before continuing then` so at least the next person who looks at it has an idea what it is about.

Reply:
https://golang.org/pkg/runtime/#GC
says 
> GC runs a garbage collection and blocks the caller until the garbage collection is complete. It may also block the entire program.

Reply:
It might say that, I've seen unpredictable things happen when not using the sleep call during other tests. That's why I added it.

Reply:
I re-tested it and it seems fine now, so I removed it.